### PR TITLE
Improve jqXHR documentation

### PR DESCRIPTION
**Pull request by gibson042**
Fixes gh-1019

### CODE REVIEW

File: entries/jQuery.ajax.xml [hunk 1]
```diff
@@ -281,24 +281,31 @@ jqxhr.always(function() {
...
         <code>statusText</code>
       </li>
-      <li><code>responseXML</code> and/or <code>responseText</code> when the underlying request responded with xml and/or text, respectively</li>
-      <li><code>setRequestHeader(name, value)</code> which departs from the standard by replacing the old value with the new one rather than concatenating the new value to the old one</li>
       <li>
-        <code>getAllResponseHeaders()</code>
+        <code>abort( [ statusText ] )</code>
+      </li>
+      <li>
+        <code>getAllResponseHeaders()</code> as a string
```
Comment (dmethvin) [inline entries/jQuery.ajax.xml]: Is it worth adding "as a string" here? The others don't talk about return types in general.

Comment (gibson042) [inline entries/jQuery.ajax.xml]: Absent prior knowledge, I'd expect `getAllResponseHeaders()` to return a map. This clarifies that to not be the case.

**Comment by AurelioDeRosa** [comment]
Should I keep all the commits here or should I fix up them?

**Comment by gibson042** [comment]
They'll always be distinct here; I'm not opposed to a landing them on master as a single unit.

**Comment by AurelioDeRosa** [comment]
Yeah, my question was basically if there was any preference on how to merge them in master.

**Comment by gibson042** [comment]
I personally prefer the squashed representation, and that's also our standard in Core and QUnit.

**Comment by mgol** [comment]
Is anything blocking landing this?

**Comment by kswedberg** [comment]
@mgol Not that I'm aware of. LGTM

**Pull request closed without merging**